capsules/src/usb/usbc_client_ctrl.rs

```diff
@@ -22,58 +34,60 @@ use core::cmp::min;
 use kernel::hil;
 use kernel::hil::usb::TransferType;
 
-const DESCRIPTOR_BUFLEN: usize = 64;
+const DESCRIPTOR_BUFLEN: usize = 128;
 
 const N_ENDPOINTS: usize = 3;
 
-pub struct ClientCtrl<'a, 'b, C: 'a> {
-    // The hardware controller
-    controller: &'a C,
+/// Handler for USB control endpoint requests.
+pub struct ClientCtrl<'a, 'b, U: 'a> {
+    /// The USB hardware controller.
+    controller: &'a U,
 
-    // State for tracking each endpoint
+    /// State of each endpoint.
     state: [Cell<State>; N_ENDPOINTS],
 
-    // A 64-byte buffer for the control endpoint
+    /// A 64-byte buffer for the control endpoint to be passed to the USB
+    /// driver.
     ctrl_buffer: Buffer64,
 
-    // Storage for composing responses to device-descriptor requests
+    /// Storage for composing responses to device descriptor requests.
     descriptor_storage: [Cell<u8>; DESCRIPTOR_BUFLEN],
 
-    // Descriptors to reply to control requests
-    device_descriptor: DeviceDescriptor,
-
-    // For now we only support one configuration...
-    configuration_descriptor: ConfigurationDescriptor,
+    /// Buffer containing the byte-packed representation of the device
+    /// descriptor. This is expected to be created and passed from the user of
+    /// `ClientCtrl`.
+    device_descriptor_buffer: DeviceBuffer,
 
-    // ...with only one interface
-    interface_descriptor: InterfaceDescriptor,
+    /// Buffer containing the byte-packed representation of the configuration
```

Comment:
What does "byte-packed" mean? Do you mean "serialized"? I've heard of "bit packing" before, but never of "byte packing".

Reply:
I mean it in this context https://docs.python.org/3/library/struct.html

Reply:
Well, there is no Python in this context, so it's not clear here.

Reply:
I don't think  byte-packed is confusing, but we could just throw the word serialized in instead.

```suggestion
    /// Buffer containing the byte-serialized representation of the configuration
```